src/dbnode/encoding/series_iterator.go

```diff
@@ -101,10 +97,17 @@ func (it *seriesIterator) Close() {
 		return
 	}
 	it.closed = true
-	it.id.Finalize()
-	it.nsID.Finalize()
+	if it.id != nil {
```

Comment:
What changed that we suddenly needed these nil checks?

Reply:
No new change, I'm just making sure we do the same nil check for all fields that are nil-able here.